serving/src/main/java/ai/djl/serving/wlm/WorkLoadManager.java

```diff
@@ -1,119 +1,48 @@
-/*
- * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
- *
- * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
- * with the License. A copy of the License is located at
- *
- * http://aws.amazon.com/apache2.0/
- *
- * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
- * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
- * and limitations under the License.
- */
 package ai.djl.serving.wlm;
 
-import ai.djl.serving.util.ConfigManager;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.List;
+import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executor;
 import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
-import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.SynchronousQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
 
-class WorkLoadManager {
+public class WorkLoadManager {
 
-    private ConfigManager configManager;
-    private AtomicInteger gpuCounter;
-    private ExecutorService threadPool;
-    private ConcurrentHashMap<String, List<WorkerThread>> workers;
+    private Executor defaultPredictionExecutor;
+    private Map<String, ExecutorService> specialisedPredictionExecutor;
 
-    public WorkLoadManager(ConfigManager configManager) {
-        this.configManager = configManager;
-        this.gpuCounter = new AtomicInteger(0);
-        threadPool = Executors.newCachedThreadPool();
-        workers = new ConcurrentHashMap<>();
+    public WorkLoadManager() {
+        specialisedPredictionExecutor = new ConcurrentHashMap<>();
+        defaultPredictionExecutor =
```

Comment:
My idea was to use 1 PredictionExecutor for all parallel executors. Except you need a special prioritized behaviour for a specific model. Otherwise you do not load balance correctly when you have x models loaded and you get requests for different models at the same time then you run them all in parallel. But you just have limited CPU-resources. Let's say you want to have 1 currently-running-parallel thread per CPU Core.  but you have workers which are idle for one model, another model uses all the model-specific workers....